README.md

```diff
@@ -129,6 +129,111 @@ It is _critical_ that you require the user to be logged in and fetch the
 user identifier and email (e.g. `$user->getid()` and `$user->getEmail()`)
 from that authenticated user (not from anywhere in the URL).
 
+## Anonymous Validation
+
+It is also possible to allow users to verify their email address without having
+to be authenticated. A use case for this would be if a user registers on their laptop,
+but clicks the verification link on their phone. Normally, the user would be
+required to log in before their email was verified. 
+
+We can overcome this by passing a user identifier as a query parameter in the
+signed url. The examples below demonstrate how this is done.
+
+```php
+// RegistrationController.php
+
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Component\Mailer\MailerInterface;
+use SymfonyCasts\Bundle\VerifyEmail\VerifyEmailHelperInterface;
+...
+
+class RegistrationController extends AbstractController
+{
+    private $verifyEmailHelper;
+    private $mailer;
+    
+    public function __construct(VerifyEmailHelperInterface $helper, MailerInterface $mailer)
+    {
+        $this->verifyEmailHelper = $helper;
+        $this->mailer = $mailer;
+    }
+    
+    /**
+     * @Route("/register", name="register-user")
+     */
+    public function register(): Response
+    {
+        $user = new User();
+    
+        // handle the user registration form and persist the new user...
+    
+        $signatureComponents = $this->verifyEmailHelper->generateSignature(
+                'registration_confirmation_route',
+                $user->getId(),
+                $user->getEmail(),
+                ['id' => $user->getId()] // add the users id as an extra query param
+            );
+        
+        $email = new TemplatedEmail();
+        $email->to($user->getEmail());
+        $email->htmlTemplate('registration/confirmation_email.html.twig');
+        $email->context(['signedUrl' => $signatureComponents->getSignedUrl()]);
+        
+        $this->mailer->send($email);
+    
+        // generate and return a response for the browser
+    }
```

Comment:
hmm, from a maintenance perspective (and for better clarity, though also worse copy-and-paste ability), could we use a `diff` code block? And use the + and - lines that would need to *change* when you want to allow this feature? Instead of showing all the code.

Reply:
genius!